clippy_lints/src/needless_question_mark.rs

```diff
@@ -0,0 +1,191 @@
+use rustc_errors::Applicability;
+use rustc_hir::def::{CtorKind, CtorOf, DefKind, Res};
+use rustc_hir::{Body, Expr, ExprKind, LangItem, MatchSource, QPath};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::DefIdTree;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::sym;
+
+use crate::utils;
+use if_chain::if_chain;
+
+declare_clippy_lint! {
+    /// **What it does:**
+    /// Suggest for terminating expressions
+    /// ```rust,ignore
+    /// value.inner_option;
+    /// value.inner_result;
+    /// ```
+    /// instead of
+    /// ```rust,ignore
+    /// Some(value.inner_option?);
+    /// Ok(value.inner_result?);
+    /// ```
```

Comment:
This should be in the examples. The "What it does" section is to describe in words what the lint does.